### DISCUSSION TITLE
How do I read a TGA file?

### DISCUSSION THREAD

**Discussion by ghost** [dlemstra/magick-wasm#59] [Q&A] [answered]
Hi. I'm using Imagemagick.read() to load a file:
` 
const a = new Promise<Uint8Array>((resolve) => {
      magick.read(view,(img) => {
        img.write((data) => {
          resolve(data)
          img.dispose()
        }, f)
      })
    })
` Loading a JPG or PNG file works great. The problem is, sometimes I also have to load a TGA file, but they refuse to work:
`Error: NoDecodeDelegateForThisImageFormat ``' @ error/blob.c/BlobToImage/460`.
So what's the problem? Am I doing something wrong? Can I force the type of input to read()?

**Comment by dlemstra** [comment, marked as answer]
For TGA you will need to specify the format in the `MagickReadSettings` and use the correct `read` overload.
